Harmony/Internal/Util/CecilEmitter.cs

```diff
@@ -129,4 +157,15 @@ private static string SanitizeTypeName(string typeName)
 			.Replace("<", "{")
 			.Replace(">", "}");
 	}
+
+	private static Instruction LabelFix(ILLabel label, Collection<Instruction> clone, Collection<Instruction> original)
+	{
+		var target = label.Target;
+		if (!clone.Contains(target))
+		{
+			var idx = original.IndexOf(label.Target);
+			if (idx != -1) return clone[idx];
+		}
+		return target;
+	}
```

Comment:
Are `clone` and `original` guaranteed to have the same count? If so, add a `Debug.Assert` call. If not, `clone[idx]` is a bug.

Are there actually situations where `clone.Contains(target)` returns true?

I'm uncomfortable with ever returning `target`. I feel like it should throw if the instruction isn't found.